services/controller/standard/events.go

```diff
@@ -332,3 +339,61 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
 	s.subscriptionInfos[epoch] = subscriptionInfo
 	s.subscriptionInfosMutex.Unlock()
 }
+
+// VerifySyncCommitteeMessages handles the "head" events from the beacon node for sync committee validation.
+func (s *Service) VerifySyncCommitteeMessages(ctx context.Context, data *apiv1.HeadEvent) {
+	_, span := otel.Tracer("attestantio.vouch.services.controller.standard").Start(ctx, "VerifySyncCommitteeMessages")
+	defer span.End()
+
+	// We verify against the previous slot as that is when the sync committee will have reported.
+	previousSlot := data.Slot - 1
+	currentSlot := data.Slot
+
+	// Logging with the current slot as that is when this code is executed.
+	log := log.With().
+		Uint64("current_slot", uint64(currentSlot)).
+		Uint64("previous_slot", uint64(previousSlot)).
+		Str("block_header", data.Block.String()).
+		Logger()
+	log.Trace().Msg("Received head event")
+
+	previousSlotData, found := s.syncCommitteeMessenger.GetDataUsedForSlot(previousSlot)
+	if !found {
+		log.Trace().Msg("No reported sync committee message data for slot; skipping validation")
+		return
+	}
+	blockResponse, err := s.signedBeaconBlockProvider.SignedBeaconBlock(ctx, &api.SignedBeaconBlockOpts{
```

Comment:
Fetching the block is relatively heavyweight compared to just fetching the block header, which should be all that is needed here.

Reply:
I had a quick look at the `beaconBlockHeadersProvider` and couldn't see any way to get the `SyncAggregate` data for us to check if our validators were included. Is there something I'm missing or a lighter way to get the `SyncAggregate` data?

Reply:
Yep my bad I didn't look past the use of the block for the parent root.